### PR TITLE
Fixes #27162 - Use roles_path from settings

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by xprazak2** [comment]
I fixed the tests.

**Comment by ares** [comment]
needs https://github.com/theforeman/smart_proxy_ansible/pull/19

File: lib/smart_proxy_ansible/roles_reader.rb [hunk 1]
```diff
@@ -33,6 +30,14 @@ def logger
 
         private
 
+        def default_roles_paths
+          ::Proxy::Ansible::Plugin.settings.ansible_roles_path
+        end
+
+        def default_config_file
```
[the remark this answers is not in the document]
Comment (xprazak2) [reply lib/smart_proxy_ansible/roles_reader.rb]: There is not a non-default. I just kept the same name as the constant had.

**Comment by xprazak2** [comment]
I made changes to always rely on `roles_path` that we deploy into `ansible.cfg` as suggested in changes for [installer](https://github.com/theforeman/puppet-foreman_proxy/pull/518/files#r298210501)

File: lib/smart_proxy_ansible/roles_reader.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (xprazak2) [reply lib/smart_proxy_ansible/roles_reader.rb]: Now it is renamed

**Comment by adamruzicka** [comment]
Needs a rebase

File: lib/smart_proxy_ansible/roles_reader.rb [hunk 2]
```diff
@@ -33,6 +30,10 @@ def logger
 
         private
 
+        def config_file_path
+          File.join ::Proxy::Ansible::Plugin.settings.ansible_working_dir, '.ansible.cfg'
```
Comment (ezr-ondrej) [inline lib/smart_proxy_ansible/roles_reader.rb:34]: this is wrong. In the working_directory we should look for `/ansible.cfg` according to https://docs.ansible.com/ansible/latest/reference_appendices/config.html
And shouldn't we acrually implement a fallback to the default? :thinking:

The fact that working dir is by default == to home dir should not IMO matter, user can change that and get confused or we can decide to change the default in the future.
I think we should go with `<working_directory>/ansible.cfg` with fallback to `~<rex/ansible user>/.ansible.cfg` and maybe even fallback to the system config.
Do you think we can do something like that ? :thinking:

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]
Looks good, I've rebased to see the tests, but found the naming inconsistency and what about the config change? Won't the users who doesn't have the config in working dir get confusing results, because import will stop working for them?

**Comment by ares** [comment]
any update on this? a user asked for this recently (well they wanted to scan roles directory recursively, but having a way to set the roles path would be much better and safer)

**Comment by stejskalleos** [comment]
Closing the PR, it's been two years since the last response.

**Pull request closed without merging**